### PR TITLE
replace plain error with DimensionMismatch

### PR DESCRIPTION
**Pull request by matbesancon**

### CODE REVIEW

**Comment by matbesancon** [comment]
Some boundcheck propagation and tagging were also missing

File: src/MArray.jl [hunk 1]
```diff
@@ -134,7 +134,7 @@ macro MArray(ex)
             s2s = map(i -> ((isa(ex.args[i], Expr) && ex.args[i].head == :row) ? length(ex.args[i].args) : 1), 1:s1)
             s2 = minimum(s2s)
             if maximum(s2s) != s2
-                error("Rows must be of matching lengths")
+                throw(DimensionMismatch("Rows must be of matching lengths"))
```
Comment (c42f) [inline src/MArray.jl]: I feel like the malformed AST which would be passed in this case is more an `ArgumentError` to the macro than a `DimensionMismatch`.  The same could be said of the other macros.

**Comment by matbesancon** [comment]
just updated the errors in macros

**Comment by c42f** [comment]
Thanks, I think there's a few you missed.

Eh, we have too many of these macros :-/

**Comment by c42f** [comment]
There, I went and made those consistent as well. Let's check CI again.

**Comment by matbesancon** [comment]
Thanks! CI is passing, but I doubt the errors were covered

**Pull request merged**